packages/@aws-cdk/core/lib/secret-value.ts

```diff
@@ -76,9 +88,11 @@ export class SecretValue extends Intrinsic {
    * If possible, use `SecretValue.ssmSecure` or `SecretValue.secretsManager` directly.
    *
    * @param ref The dynamic reference to use.
+   *
+   * @param secretArn The secretArn if the reference is for a secret.
    */
-  public static cfnDynamicReference(ref: CfnDynamicReference) {
-    return new SecretValue(ref);
+  public static cfnDynamicReference(ref: CfnDynamicReference, secretArn?: string) {
```

Comment:
Let's leave this as-is (I actually think adding this parameter here doesn't make much sense), and instead change the internal callers of `cfnDynamicReference` to just use `new SecretValue(...)` directly.